packages/block-library/src/calendar/index.php

```diff
@@ -13,7 +13,12 @@
  * @return string Returns the block content.
  */
 function render_block_core_calendar( $attributes ) {
-	global $monthnum, $year;
+	global $monthnum, $year, $wpdb;
+
+	$has_posts = $wpdb->get_var( "SELECT 1 as test FROM $wpdb->posts WHERE post_type = 'post' AND post_status = 'publish' LIMIT 1" );
```

Comment:
This is a pretty lean query but most of the time we want to cache values like this (in an option or a transient) to avoid hammering the DB (we have tons of users on underpowered shared hosts). That would then require invalidating the cache on every `update_post` action. 

This is just what's done, by default, with the `post_count` option, but unfortunately it only works on Multisite setups. (See `update_post_count()`.)

I don't think this is a blocker, but we'd probably wind up needing to do it sooner or later anyway.

Reply:
> Thank you for the review!

Agreed! Thanks @mattwiebe TIL.